buildroot-external/board/hardkernel/patches/linux/0001-arm64-dts-meson-add-RTC-to-ODROID-N2-boards.patch

```diff
@@ -1,50 +0,0 @@
-From 516c081bcdd8f248982588aa8ca5bd8d24b2e7ef Mon Sep 17 00:00:00 2001
-Message-Id: <516c081bcdd8f248982588aa8ca5bd8d24b2e7ef.1612541102.git.stefan@agner.ch>
-From: Stefan Agner <stefan@agner.ch>
-Date: Mon, 16 Nov 2020 23:11:02 +0100
-Subject: [PATCH 1/4] arm64: dts: meson: add RTC to ODROID-N2 boards
-
-All ODROID-N2 boards come with a NXP PCF8563TS RTC connected to I2C bus
-3. This is the RTC which is connected to the on-board RTC backup battery.
-Use it as primary RTC.
-
-Signed-off-by: Stefan Agner <stefan@agner.ch>
----
- .../boot/dts/amlogic/meson-g12b-odroid-n2.dtsi     | 14 ++++++++++++++
- 1 file changed, 14 insertions(+)
-
-diff --git a/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2.dtsi b/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2.dtsi
-index 39a09661c5f6..445d90d25aa3 100644
---- a/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2.dtsi
-+++ b/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2.dtsi
-@@ -13,6 +13,8 @@ / {
- 	aliases {
- 		serial0 = &uart_AO;
- 		ethernet0 = &ethmac;
-+		rtc0 = &board_rtc;
-+		rtc1 = &vrtc;
- 	};
- 
- 	dioo2133: audio-amplifier-0 {
-@@ -471,6 +473,18 @@ hdmi_tx_tmds_out: endpoint {
- 	};
- };
- 
-+&i2c3 {
-+	status = "okay";
-+	pinctrl-0 = <&i2c3_sda_a_pins>, <&i2c3_sck_a_pins>;
```

Comment:
I prefer this patch over the hacked up patches below since it makes use of existing pinctrl's. I am planning to upstream this patch.

Can you leave this in, and rebase the other ones on this one (e.g. remove i2c3 and just add i2c2?

Reply:
As stated in the PR description, I have largely borrowed the kernel patches for the hardkernel/odroid platform from the armbian project (see https://github.com/armbian/build/tree/master/patch/kernel/archive/meson64-5.10). And as the proposed patches are dependent on eachother I decided to drop your patch in favour of the official armbian-based patchsets. However, if you insist I could of course see if I can keep your patch and rework the others. But then they are not compatible to the armbian patches anyone and thus might be harder to be updated in future in case armbian integrates more changes.

Reply:
> official armbian-based patchsets

Often those are downstream patches, and I already inherited breaking changes.

"Official" in my eyes is whatever is upstream. And if there are upstream patches available which do the same thing, I prefer those since I can just delete them once we move to newer kernel.

In this case there seem to be patches already merged at least for I2C3/RTC:
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/log/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2.dtsi

So this should be a simple cherry-pick.

Reply:
Since I am currently working on ODROID-N2 anyways I can prepare a patchset today which should be an equivalent change and then you can rebase your changes on-top. Is that ok for you?

Reply:
That's perfect with me, thanks.